### PR TITLE
HC-9-Events-model - create table in to aws

### PR DESCRIPTION
**Pull request by JaCoWymowny**

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@types/express": "^4.17.17",
     "@types/node": "^18.15.11",
     "@types/sequelize": "^4.28.14",
+    "@types/validator": "^13.7.15",
```
[the remark this answers is not in the document]
Comment (JaCoWymowny) [reply server/package.json:27]: I haven't used it anywhere yet, so I'll remove it

File: server/src/app.ts [hunk 2]
```diff
@@ -1,3 +1,5 @@
+import "reflect-metadata";
```
[the remark this answers is not in the document]
Comment (JaCoWymowny) [reply server/src/app.ts:1]: This is required by the sequelize-typescript to enable automatic mapping between model classes and database tables. reflect-metadata provides information about types and class decorators that are required by sequelize-typescript.

import "reflect-metadata" allows you to use decorators in TypeScript/JavaScript code and use the Reflection mechanism at runtime. When using decorators, the "reflect-metadata" import must be called before defining the decorators in the code. The Reflection mechanism, on the other hand, allows programmers to obtain information about types, attributes and functions in the code while the program is running.

We may need to use the second part, which is Reflection (Reflect.method), when expanding the application. However, to ensure that the application does not crash when using decorators, it is necessary to use import "reflect-metadata"; in some places. Thanks to delving into the topic, I noticed that I didn't import that phrase into the models.ts file.

File: server/src/models/events.model.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Model, Table, Column, DataType } from "sequelize-typescript";
+
+@Table({ tableName: "events" })
+export class Event extends Model {
```
[the remark this answers is not in the document]
Comment (JaCoWymowny) [reply server/src/models/events.model.ts:4]: This is the definition of a sample table I was supposed to create for version 1.0 of the project. This is how I create a table definition with columns, extending the MODEL class with the appropriate data along with the types. After initializing the connection to the database, I can create properly named tables by importing them from each `xxx.models.ts` file separately, maintaining code consistency. At a later stage, it is verified whether the table is created with the default `force: false` setting, so as not to reset existing data. In addition, a model file written in this way is easily editable and allows extending the functionality of a given table, in accordance with the rules of using the sequeilize-typescript library


ID, createdAT, updatedAT fields are created automatically
image

File: server/src/models/events.model.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import "reflect-metadata";
+import { Model, Table, Column, DataType } from "sequelize-typescript";
+
+@Table({ tableName: "events" })
+export class Event extends Model {
+  @Column({
+    type: DataType.STRING,
+    allowNull: false,
+  })
+  eventName!: string;
```
Comment (kj-ninja) [inline server/src/models/events.model.ts:10]: why are you using ! here?

File: server/src/models/events.model.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+  @Column({
+    type: DataType.DATEONLY,
+    allowNull: false,
+  })
+  eventDate!: Date;
```
Comment (kj-ninja) [inline server/src/models/events.model.ts:16]: add one more field like description

**Review by kj-ninja — APPROVED** [review]

File: server/src/models/events.model.ts (hunk 4 above)
Comment (JaCoWymowny) [reply server/src/models/events.model.ts:10]: This is to accurately type the data and eliminate nulls and undefineds from the table. I think that's a good proposition. The guides do suggest using this, but it's not a requirement to push the project forward.

**Pull request merged**